PennyMe/MachineChangedRequest.swift

```diff
@@ -0,0 +1,333 @@
+//
+//  MachineChangedRequest.swift
+//  PennyMe
+//
+//  Created by Nina Wiedemann on 04.11.23.
+//  Copyright © 2023 Jannis Born. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+import MapKit
+import Combine
+
+import SwiftUI
+import MapKit
+import CoreLocation
+
+// variable defining how large the shown region is when changing coordinates
+let regionInMeters: Double = 100
+
+@available(iOS 14.0, *)
+struct MapView: View {
+    @State private var region: MKCoordinateRegion
+    @State private var showDoneAlert = false
+    @Binding private var centerCoordinate: CLLocationCoordinate2D
+    @Environment(\.presentationMode) private var presentationMode
+    
+    init(centerCoordinate: Binding<CLLocationCoordinate2D>, initialCenter: CLLocationCoordinate2D) {
+            _centerCoordinate = centerCoordinate
+            let regionTemp = MKCoordinateRegion(
+                center: initialCenter,
```

Comment:
Here, I can't initialize the MKCoordinateRegion center with a `Binding` variable, so I need to pass the coordinates to initialize the region. This has the effect that the map is always resetted to the original machine coordinates when re-opening the map.